share/functions/__fish_print_xdg_applications_directories.fish

```diff
@@ -0,0 +1,10 @@
+function __fish_print_xdg_applications_directories --description 'Print directories where desktop files are stored'
+    set -l search_path ~/.local/share/applications /usr/share/applications
+    if test -d /usr/local/share/applications
```

Comment:
True this should probably be done for all paths. `__fish_print_xdg_mimetypes` redirects finds stderr to `/dev/null` but `__fish_print_xdg_mimeapps` doesn't.  This could course error massages in the output.